src/petals/server/block_functions.py

```diff
@@ -72,8 +77,12 @@ async def run_rpc_backward(
     active_adapter: str = "",
     prioritizer: TaskPrioritizerBase,
     points: int = 0,
+    structure: Optional[Dict[str, Any]] = None,
 ) -> Union[torch.Tensor, Sequence[torch.Tensor]]:
-    inputs, grad_outputs, prompts = flat_tensors
+    if structure is not None:
+        flat_tensors, kwargs = unpack_args_kwargs(flat_tensors, structure)
```

Comment:
If kwargs are not used intentionally (for this PR), please leave a TODO to actually forward them